<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>973</width>
    <height>641</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,3">
    <property name="leftMargin">
     <number>9</number>
    </property>
    <property name="topMargin">
     <number>9</number>
    </property>
    <property name="rightMargin">
     <number>9</number>
    </property>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0,4,1">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="toolbar" native="true">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="addGroupBtn">
            <property name="minimumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="toolTip">
             <string>Add group</string>
            </property>
            <property name="text">
             <string>+ G</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addControlBtn">
            <property name="minimumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="toolTip">
             <string>Add control</string>
            </property>
            <property name="text">
             <string>+ C</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addParamBtn">
            <property name="minimumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="toolTip">
             <string>Add parameter</string>
            </property>
            <property name="text">
             <string>+ P</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="removeBtn">
            <property name="minimumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>41</width>
              <height>26</height>
             </size>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="toolTip">
             <string>Remove element</string>
            </property>
            <property name="text">
             <string>--</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>47</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QTreeWidget" name="controlsTree">
         <attribute name="headerVisible">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string notr="true">1</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <widget class="QWidget" name="elemPropsWidget" native="true"/>
       </item>
      </layout>
      <zorder>controlsTree</zorder>
      <zorder>toolbar</zorder>
      <zorder>elemPropsWidget</zorder>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget_2" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="2,1">
       <item>
        <widget class="QScrollArea" name="controlsView">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>692</width>
            <height>367</height>
           </rect>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="commandsLogView">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>973</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="openXmlAction"/>
    <addaction name="openJsonAction"/>
    <addaction name="saveXmlAction"/>
    <addaction name="saveJsonAction"/>
    <addaction name="copyXmlAsCVarAction"/>
    <addaction name="copyJsonAsCVarAction"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="openXmlAction">
   <property name="text">
    <string>Open xml</string>
   </property>
  </action>
  <action name="saveJsonAction">
   <property name="text">
    <string>Save as json</string>
   </property>
  </action>
  <action name="saveXmlAction">
   <property name="text">
    <string>Save as xml</string>
   </property>
  </action>
  <action name="copyXmlAsCVarAction">
   <property name="text">
    <string>Copy xml to buffer as C var</string>
   </property>
  </action>
  <action name="copyJsonAsCVarAction">
   <property name="text">
    <string>Copy json to buffer as C var</string>
   </property>
  </action>
  <action name="openJsonAction">
   <property name="text">
    <string>Open json</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
